src/main/java/com/google/devtools/build/lib/analysis/starlark/StarlarkTransition.java

```diff
@@ -381,6 +338,74 @@ private static Target getActual(
     return cleanedOptionMap.buildOrThrow();
   }
 
+  /**
+   * Validate the value of a particular build setting after a transition has been applied.
+   * @param buildSettingRule the build setting to validate.
+   * @param options the {@link BuildOptions} reflecting the post-transition configuration.
+   * @param maybeAliasSetting the label used to refer to the build setting in the transition,
+   *                          possibly an alias. This is only used for error messages.
+   * @param inputAndOutputSettings the transition input and output settings. This is only used for
+   *                               error messages.
+   * @return {@code true} if and only if the setting is set to its default value after the
+   *         transition.
+   * @throws TransitionException if the value returned by the transition for this setting has an
+   *                             invalid type.
+   */
+  private static boolean validateAndCheckIfAtDefault(Rule buildSettingRule, BuildOptions options,
```

Comment:
Is this method pretty much a boiler plate copy of the original logic, or are there specific changes I should pay attention to?